hazelcast/src/main/java/com/hazelcast/map/AbstractEntryProcessor.java

```diff
@@ -71,4 +75,16 @@ public void processBackup(Map.Entry<K, V> entry) {
             process(entry);
         }
     }
+
+    private class HazelcastInstanceAwareEntryBackupProcessorImpl extends EntryBackupProcessorImpl
+            implements HazelcastInstanceAware {
+
+        @Override
+        public void setHazelcastInstance(HazelcastInstance hazelcastInstance) {
+            final AbstractEntryProcessor<K, V> outer = AbstractEntryProcessor.this;
+            if (outer instanceof HazelcastInstanceAware) {
```

Comment:
Actually I remember from the commit message, the first version checked if the AbstractEntryProcessor and returned a different EntryBackupProcessorImpl. So the check in the "HazelcastInstanceAwareEntryBackupProcessorImpl" didn't need to check the outer type again.